src/validation.cpp

```diff
@@ -3092,6 +3092,39 @@ void CChainState::ProcessLoanEvents(const CBlockIndex* pindex, CCustomCSView& ca
         cache.TransferVaultInterest(auction.vaultId, pindex->nHeight, vault->schemeId, {});
         return true;
     }, {CVaultId{}, static_cast<uint32_t>(pindex->nHeight)});
+
+    auto priceHeight = pindex->nHeight + Params().GetConsensus().blocksPriceUpdate();
```

Comment:
Should price height here just be the `blocksPriceUpdate`? Because you are using to modulo later `(pindex->nHeight - collateralToken->creationHeight) % priceHeight`